docs/integrations/github/discovery.md

```diff
@@ -6,6 +6,120 @@ sidebar_label: Discovery
+```
+
+And then add the entity provider to your catalog builder:
+
+```diff
+  // In packages/backend/src/plugins/catalog.ts
++ import { GitHubEntityProvider } from '@backstage/plugin-catalog-backend-module-github';
+
+  export default async function createPlugin(
+    env: PluginEnvironment,
+  ): Promise<Router> {
+    const builder = await CatalogBuilder.create(env);
++   builder.addEntityProvider(
++     GitHubEntityProvider.fromConfig(env.config, {
++       logger: env.logger,
++       schedule: env.scheduler.createScheduledTaskRunner({
++         frequency: { minutes: 30 },
++         timeout: { minutes: 3 },
++       }),
++     }),
++   );
+
+    // [...]
+  }
+```
+
+## Configuration
+
+To use the discovery provider, you'll need a GitHub integration
+[set up](locations.md) with either a [Personal Access Token](../../getting-started/configuration.md#setting-up-a-github-integration) or [GitHub Apps](./github-apps.md).
+
+Then you can add a github config to the catalog providers configuration:
+
+```yaml
+catalog:
+  providers:
+    github:
+      # the provider ID can be any camelCase string
+      providerId:
+        organization: 'backstage' # string
+        catalogPath: '/catalog-info.yaml' # string
+        filters:
+          branch: 'main' # string
+          repository: '.*' # Regex
+      customProviderId:
+        organization: 'new-org' # string
+        catalogPath: '/custom/path/catalog-info.yaml' # string
+        filters: # optional filters
+          branch: 'develop' # optional string
+          repository: '.*' # optional Regex
+```
+
+This provider supports multiple organizations via unique provider IDs
+
+> **Note:** It is possible but certainly not recommended to skip the provider ID level.
+> If you do so, `default` will be used as provider ID.
+
+- **catalogPath** _(optional)_:
+  Default: `/catalog-info.yaml`.
+  Path where to look for `catalog-info.yaml` files.
+  When started with `/`, it is an absolute path from the repo root.
+- **filters** _(optional)_:
+  - **branch** _(optional)_:
+    String used to filter results based on the branch name.
+  - **repository** _(optional)_:
+    Regular expression used to filter results based on the repository name.
+- **organization**:
+  Name of your organization account/workspace.
+  If you want to add multiple organizations, you need to add one provider config each.
+
+## GitHub API Rate Limits
+
+GitHub [rate limits](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting) API requests to 5,000 per hour (or more for Enterprise
+accounts). The default Backstage catalog backend refreshes data every 100
+seconds, which issues an API request for each discovered location.
+
+This means if you have more than ~140 catalog entities, you may get throttled by
+rate limiting. You can change the refresh frequency of the catalog in your `packages/backend/src/plugins/catalog.ts` file:
```

Comment:
I think you refer to two different time settings. Above you mention refreshes every 100 seconds, but your example config which is also related to the scheduler frequency below is about every 35 minutes.

I guess you meant the internal processing loop above. This can be configured at the catalog.ts file, too, directly at the CatalogBuilder. The snippet below though is only about the discovery schedule.

Reply:
updated for clarity.